src/cmd_all/src/bin/risingwave.rs

```diff
@@ -98,7 +99,14 @@ enum Component {
     Compactor,
     Ctl,
     Playground,
+    /// Used by cloud to bundle different components into a single node.
+    /// It exposes the low level configuration options of each node.
     Standalone,
+    /// Used by users to run a single node.
+    /// The low level configuration options are hidden.
+    /// We only expose high-level configuration options,
+    /// which map across multiple nodes.
+    SingleNode,
```

Comment:
The distinction of these two still sound confusing. ("standalone" sounds actually more high-level than "single node" to me 🤣)

Can we just e.g., change the default config of "standalone" mode to that?

Reply:
I don't understand why we need a new "mode". For quick start purpose, it doesn't need to be configurable; for advanced usage, low-level config are better.

Reply:
> Can we just e.g., change the default config of "standalone" mode to that?

1. While we can fix some default config for standalone mode for `single_node` users, these configurations may not apply to cloud / docker.
2. We still need a high level configuration layer for the users. Standalone mode does not support that well, since it is not well-designed for that. The initial goal for it is for cloud to configure different services in a single process. So we expose node level options. In the case of `single_node`, we do not need to expose these node level options.

Reply:
> or quick start purpose, it doesn't need to be configurable
Hmm but later users will still want to configure it, they may not need to go all the way to advanced usage.
But may want to reconfigure simple things like:
- Where to store data.
- Node addresses.

Reply:
> While we can fix some default config for standalone mode for single_node users, these configurations may not apply to cloud / docker.

Since `standalone` mode is configurable, we can just let cloud/docker specify their own default config?

>> or quick start purpose, it doesn't need to be configurable
>
> Hmm but later users will still want to configure it, they may not need to go all the way to advanced usage.
> But may want to reconfigure simple things like:
> - Where to store data.
> - Node addresses.

For these simple configs, it could be possibly solved by providing docs for commonly used parameters (for `standalone` mode)?

> We still need a high level configuration layer for the users. Standalone mode does not support that well, since it is not well-designed for that.

Could you give some examples that compares "low level" config vs "high level" config to demonstrate how it could provide better UX?

Reply:
If possible, I'll like to rename the original standalone mode to "multiple" or something similar, which indicates it's just designed for running multiple components in one process. 

Of course, as the existing standalone mode has been used by cloud and some of our customers, this change must keep backward compatibility. 😫

Reply:
Maybe we can hide standalone mode in CLI with `#[clap(subcommand, hide(true))]`.

Reply:
At the same time, I also feel that maybe `playground` can also be deprecated in favor of `single-node` mode...

Reply:
> If possible, I'll like to rename the original standalone mode to "multiple" or something similar, which indicates it's just designed for running multiple components in one process.

If "standalone" becomes "multiple", then "single" will be even more confusing🤔

Reply:
> Maybe we can hide standalone mode in CLI with #[clap(subcommand, hide(true))].

Unfortunately, we still need it 🤣 `./risingwave single_node --help`.

We can't fetch the help otherwise, since it will conflict with the top level's `--help`.

Reply:
> At the same time, I also feel that maybe `playground` can also be deprecated in favor of `single-node` mode...

Agree, but `playground` is used by the https://playground.risingwave.dev/. If someday we discontinued that, +1 for removing `risingwave playground`.

Reply:
> > Maybe we can hide standalone mode in CLI with #[clap(subcommand, hide(true))].
> 
> Unfortunately, we still need it 🤣 `./risingwave single_node --help`.
> 
> We can't fetch the help otherwise, since it will conflict with the top level's `--help`.

!!? Why hiding `standalone` may affect `single_node` or top level? 🤡

Reply:
> > At the same time, I also feel that maybe `playground` can also be deprecated in favor of `single-node` mode...
> 
> Agree, but `playground` is used by the [playground.risingwave.dev](https://playground.risingwave.dev/). If someday we discontinued that, +1 for removing `risingwave playground`.

We want to encourage users to use `single_node` in the future right? I think at least we can change the description of `playgound` to emphasize its "in memory" feature. And/or `#[clap(subcommand, hide(true))]`

Reply:
> Could you give some examples that compares "low level" config vs "high level" config to demonstrate how it could provide better UX?

Yeah sure, for example, here's how you would configure the meta-address in `standalone mode`:
```
./risingwave standalone --meta-node="--meta-address=..."
```

And in `single node mode`:
```
./risingwave --meta-addr="..."
```

Another example for state store, in `./risingwave`, we assume only local fs and sqlite are used, so we can let user just specify the directory to download:

```
./risingwave --store-directory='some_dir'
```

In contrast in standalone mode:

```
./risingwave standalone --meta-node="--backend=sql --sql-endpoint=<url> --state-store=<url>"
```

Reply:
> If possible, I'll like to rename the original standalone mode to "multiple" or something similar, which indicates it's just designed for running multiple components in one process.

The `standalone` mode is not very user friendly because it's originally meant for docker / cloud use.
Another approach we can take to unify them is to inline all node options eventually, then deprecate `standalone` mode.

Reply:
> !!? Why hiding standalone may affect single_node or top level? 🤡

I misread 🤡 🤡 🤡 I read it as single_node instead of standalone.

I will hide the `./risingwave playground` command and `standalone` as well.

Reply:
Hmm wait... Actually some users may still want to use playground mode right, since it's stateless. I suppose we can just mention it's in memory.

The convenience of invoking `./risingwave` instead of `./risingwave playground` should encourage users to use `single-node` instead, unless they want an in-memory version.